### PR TITLE
Add deepstack related changes to runner files

### LINKED ISSUE #1330: [Feature]: Support for Qwen3-VL-4B and 8B
### 🚀 The feature, motivation and pitch

Do we have the support team for Qwen3-VL

or

When can i expect this?

### Alternatives

_No response_

### Additional context

_No response_

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://github.com/vllm-project/tpu-inference/tree/main/docs), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by amanseervi**
Added deepstack related changes to  multimodal manager. This is to support the Qwen3-VL model's jax implementation.

# Description

#### **`tpu_inference/runner/multimodal_manager.py`**
  * Accepts encoder outputs as `{ "embeds": ..., "deepstack": ... }` and caches DeepStack features as a tuple inside the main `encoder_cache` to prevent cache footprint leaks.
  * `gather_mm_embeddings` concatenates DeepStack features with base embeddings along `axis=-1` before returning, preserving the runner's generic interface.

FIXES: #1330 

# Tests

Tested against the Unit tests.

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_inference/runner/tpu_runner.py [hunk 1]
```diff
@@ -560,6 +561,8 @@ def _init_inputs(self) -> None:
         if self.vllm_config.speculative_config:
             num_speculative_tokens = self.vllm_config.speculative_config.num_speculative_tokens
 
+        # mm_hash -> DeepStack outputs (per layer)
+        self.deepstack_cache: dict[str, list[jax.Array]] = {}
```
Comment (lk-chen) [inline tpu_inference/runner/tpu_runner.py]: I don't see "deepstack" in gpu_runner, I guess the tpu_runner change can/should be avoided 

https://github.com/search?q=repo%3Avllm-project%2Fvllm%20deepstack&type=code

Comment (amanseervi) [reply tpu_inference/runner/tpu_runner.py]: Yes, we can do without those. 
I started out by trying to eliminated changes in `persistent_batch_manager` by storing deepstack cache along with base embeddings. And then to avoid changes in `tpu_runner`, concatenated the deepstack features with base embeddings along last axis in `gather_mm_embeddings` (only had to make a small change in model code to accomodate this). 
Doing this btw also meant that the compilation can be done same as torchax path, so was able to remove that as well ([context](https://github.com/vllm-project/tpu-inference/pull/2685#discussion_r3280802444)). 

Also btw, now that the changes are very less, do you think we should get these mm changes merged along with model in #2685 ? It also has all the latest changes as this. If yes, I can close this.
